src/firmware_new/proto/nanopb.bzl

```diff
@@ -0,0 +1,53 @@
+load("@rules_proto//proto:defs.bzl", "proto_library", "proto_lang_toolchain", "ProtoInfo")
+load("@com_google_protobuf//:protobuf.bzl", "proto_gen")
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+def c_proto_library(name, srcs, **kwargs):
+
+    """Creates a cc_library given the proto files.
+
+    nanopb relies on protoc to do the parsing before it converts the files into
+    c header/source files. This macro runs protoc to get the desired output, and then
+    puts it through the generator script to generate the c/h files (which is then turned into a lib)
+
+    Args:
+      name: name of the filegroup that will contain compiled pb files
+      srcs: list of proto files
+
+    """
+    c_outs = []
+    h_outs = []
+
+    for src in srcs:
+
+        proto_file = src.rsplit('/', 1)[-1]
```

Comment:
Does this break if the proto is not in the same folder as the one this macro is called from? Should we use absolute paths instead?

Reply:
yeah it does break but Bazel expects you to put a BUILD file in the folder w/ the src anyway so I figured it would be easier to be relative. it might be easier to enforce that rule than have to edit this path for every proto folder

Reply:
Ok, please just make a note of this in the the ticket for the full-blown rule from the above comment.

Reply:
added